### PR TITLE
fix reversed archive ordering with --last

### PR DESCRIPTION
**Pull request by enkore**

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
we better check whether this global change has unintended effects, e.g. compared to 1.0 or concerning performance of some ops.

File: src/borg/helpers.py [hunk 1]
```diff
@@ -231,10 +231,13 @@ def list(self, sort_by=(), reverse=False, glob=None, first=None, last=None):
...
             archives.sort(key=attrgetter(sortkey))
-        if reverse or last:
+        if reverse:
             archives.reverse()
-        n = first or last or len(archives)
-        return archives[:n]
+        if first:
+            archives = archives[:first]
+        elif last:
+            archives = archives[len(archives) - last:]
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: if last is bigger than len(archives), this will get negative and will malfunction.

**Comment by enkore** [comment]
1.0 didn't have this globally.

File: src/borg/helpers.py (hunk 1 above)
Comment (enkore) [inline src/borg/helpers.py]: Good point.

**Comment by ThomasWaldmann** [comment]
yes, but for some commands.

**Comment by enkore** [comment]
Only check had that and ordering doesn't matter there for functionality.

File: src/borg/helpers.py [hunk 2]
```diff
@@ -231,10 +231,13 @@ def list(self, sort_by=(), reverse=False, glob=None, first=None, last=None):
...
-        if reverse or last:
+        if reverse:
             archives.reverse()
-        n = first or last or len(archives)
-        return archives[:n]
+        if first:
+            archives = archives[:first]
+        elif last:
+            archives = archives[max(len(archives) - last, 0):]
+        return archives
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: hmm, reverse() now needs to be done after first/last. otherwise it will turn last into first and vice versa.

Comment (enkore) [inline src/borg/helpers.py]: Yeah, the only ordering that actually makes sense is match-sort-firstlast-reverse.

Comment (ThomasWaldmann) [inline src/borg/helpers.py]: i shortly considered whether applying both first and last (or last and then first) makes sense, but guess there is no use case for that.

**Pull request merged**